sdk/identity/azure-identity/CHANGELOG.md

```diff
@@ -8,6 +8,7 @@
 
 ### Breaking Changes
 
+- Modify the order of the credentials used within the `DefaultAzureCredential` to be consistent with other languages.
 - Add `WorkloadIdentityCredential` to the `DefaultAzureCredential`.
```

Comment:
Do you have a scenario in mind that's broken by this addition? I'm struggling to come up with a plausible one, maybe due to lack of imagination 😊

Reply:
Nope, I don't have a scenario.

It is only a theoretical breaking change since the DAC order changed, and I moved it from `Feature Added` to `Breaking Change` since @LarryOsterman suggested it would be a good idea to highlight it as such, here :)
https://github.com/Azure/azure-sdk-for-cpp/pull/4940#discussion_r1323323039

Reply:
My primary concern is a customer who was previously authenticated with `AzureCliCredential` which now authenticated via `WorkloadIdentityCredential` (or some other credential).

Since the configuration for each credential could plausibly be different identities, that could potentially break customer scenarios. So changing the order of credentials in DAC is plausibly a breaking change.

If you think about it, this scenario is EXACTLY mirroring the entire purpose of this PR: We want the order of DAC to be consistent between languages because we want the behavior to be the same across languages. Why would that matter? Because the order of operations can change behavior. And thus changing the order (or adding new credential providers) is potentially a breaking change.